pasqal_cloud/client.py

```diff
@@ -176,37 +177,28 @@ def _cancel_batch(self, batch_id: str) -> Dict[str, Any]:
         )["data"]
         return response
 
-    def rebatch(
-        self,
-        batch_id: str,
-        job_ids: Union[List[str], EmptyFilter],
-        status: Union[List[str], EmptyFilter],
-        min_runs: Union[int, EmptyFilter],
-        max_runs: Union[int, EmptyFilter],
-        start_date: Union[datetime, EmptyFilter],
-        end_date: Union[datetime, EmptyFilter],
+    def _rebatch(
+        self, batch_id: Union[UUID, str], filters: RebatchFilters
     ) -> Dict[str, Any]:
-        query_params: dict[str, str | list[str]] = {}
-        if not isinstance(job_ids, EmptyFilter):
-            query_params["id"] = job_ids
-        if not isinstance(status, EmptyFilter):
-            query_params["status"] = status
-        if not isinstance(min_runs, EmptyFilter):
-            query_params["min_runs"] = str(min_runs)
-        if not isinstance(max_runs, EmptyFilter):
-            query_params["max_runs"] = str(max_runs)
-        if not isinstance(start_date, EmptyFilter):
-            query_params["start_date"] = start_date.isoformat()
-        if not isinstance(end_date, EmptyFilter):
-            query_params["end_date"] = end_date.isoformat()
-
         response: Dict[str, Any] = self._request(
             "POST",
             f"{self.endpoints.core}/api/v1/batches/{batch_id}/rebatch",
-            params=query_params,
+            params=filters.model_dump(exclude_unset=True),
         )["data"]
         return response
 
+    def _get_jobs(
```

Comment:
Would argue this isn't private either, especially if we plan on exposing it to the `__init__.py` file